src/include/stir/listmode/CListModeData.h

```diff
@@ -232,11 +233,16 @@ class CListModeData : public ExamData
       return 0;
   }
 
+  virtual
+  shared_ptr<ProjDataInfo> get_proj_data_info_sptr() const = 0;
+
 protected:
-  //! Has to be set by the derived class
+  //! Has to be initialised by the derived class
   shared_ptr<Scanner> scanner_sptr;
   //! Has to be set by the derived class
 //  shared_ptr<ExamInfo> exam_info_sptr;
+
+  shared_ptr<ProjDataInfo> proj_data_info_sptr;
```

Comment:
It doesn't make sense to have `get_proj_data_info_sptr()` pure virtual, while adding a member `proj_data_info_sptr` here. This will just lead to all derived classes having to implement `get_proj_data_info_sptr()` in exactly the same way.

Using a pure virtual `get_proj_data_info_sptr()` without having `proj_data_info_sptr` in the base class would force derived classes to store their own `proj_data_info_sptr` and will catch any problems at compilation stage, so that's good. But it will lead to code duplication. 

I therefore suggest to 
- implement `get_proj_data_info_sptr()` here (possibly with an explicit check that it's not null, not just a debug check)
- remove `scanner_sptr`, which will cause compilation problems for derived classes telling them they need to change
- ideally make `proj_data_info_sptr` private (implying we'll need a (protected) `set_proj_data_info_sptr()`)

Reply:
:ok: